### ISSUE TITLE
FPS drop

### ISSUE THREAD

**Issue by Tiounz** [Anonomit/ZeraTooltip#1]
I'm noticing a drop of 10-15 FPS when a tooltip uses ZeraTooltip rewording.

**Comment by Anonomit** [comment]
I'm thinking that some other addon may be causing more frequent than usual tooltip updates. Can you try running this script ingame? Paste it into chat as one line. It'll make a line of text appear in your chat frame whenever ZeraTooltip works on tooltips. You can reload UI or relog to undo it.
```lua
/run for _,t in ipairs{GameTooltip,ItemRefTooltip,ItemRefShoppingTooltip1,ItemRefShoppingTooltip2,ShoppingTooltip1,ShoppingTooltip2}do t:HookScript("OnTooltipSetItem",function()print"updating item tooltip"end)end
```
Typically tooltips will update a few times per second, but it's possible that an addon could be causing far more than that. Running that script and then looking at a tooltip should put about 5 lines per second in your chat frame. Let me know if you see significantly more than that. ZeraTooltip isn't doing anything more demanding than string pattern matching, but it is doing a lot of it. This should still be insignificant to performance, and I've not found any performance issues, but I could still think of at least one way to reduce processing.

Can you also make sure to test with no other addons loaded? I'm assuming that this is a conflict, and if so then it would help to know which addon(s) conflict so I can try to work around it.

**Comment by Tiounz** [comment]
Thank you for your answer, I thought it was a conflict with RatingBuster but it seems its not. I have enabled the chat timestamps to see how many tooltip updates I had per seconds and it's significantly more than 5, more like ~20, same thing goes with no other addon loaded.

I have also noticed that the FPS drop is not always as important and depends on the item I'm looking at : a cloak for example is a drop of ~15 FPS, legs are the double.

I just made a last test : I ran the script with ZeraTooltip disabled, I had almost the same number of tooltip updates, except they did not cause FPS drop.

**Comment by Anonomit** [comment]
That's strange, since my client seems to update tooltips anywhere from 3-5 times per second even with no addons enabled. I hadn't thought that this would change for other people, and I also would not have expected it to make any performance impact. I should have artificially inflated the number of tooltip updates during testing to see how it performed with hundreds of them per second, for example.

I'll see about implementing a tooltip cache. It'll mean less processing, but more memory. Thanks for letting me know about this.

**Comment by Anonomit** [comment]
Oh and as for RatingBuster, I am specifically trying to maintain compatibility with that addon. As far as I know, none of the RatingBuster info is being stripped away or adjusted on any stats.

**Comment by Anonomit** [comment]
I've implemented a cache feature in 2.0.0 (d10a54e497fc0ad2ac034a1a6fee7c8ce1007d69), this should greatly reduce processing for many tooltips. It's especially impactful when the shift key is held to show the comparison tooltips.

I also found some issues with default blizzard frames where they would ignore the tooltip update period (which is 0.2, or 5 times per second). This was causing some tooltips to update far more often (every single frame). The guilty parties were the Inspect window, the Mail window, and the Auction House. There may be more that I didn't find. ZeraTooltip now slows down tooltip updates for those windows to the normal amount of 5 per second. This should help performance for other tooltip modifying addons too.

I'm closing this issue, but please let me know in the future if you have any more performance problems with tooltips.
